src/Build/BackEnd/Components/RequestBuilder/ThreadSafeTaskRegistry.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Frozen;
+using System.Collections.Generic;
+
+#nullable disable
+
+namespace Microsoft.Build.BackEnd
+{
+    /// <summary>
+    /// Registry that maintains an allowlist of tasks that are known to be thread-safe
+    /// and can be executed within thread nodes in multi-threaded execution mode.
```

Comment:
rewrite the documentation of this file noting it is a temporary solution before we adopt a Thread safe task interface